apps/api/src/app/auth/usecases/login/login.usecase.ts

```diff
@@ -21,15 +22,77 @@ export class Login {
     const email = normalizeEmail(command.email);
     const user = await this.userRepository.findByEmail(email);
     if (!user) throw new ApiException('User not found');
+
+    if (this.isAccountBlocked(user)) {
+      throw new ApiException('Account blocked, Please try again after 5 minutes');
+    }
+
     if (!user.password) throw new ApiException('OAuth user login attempt');
 
     const isMatching = await bcrypt.compare(command.password, user.password);
-    if (!isMatching) throw new ApiException('Wrong credentials provided');
+    if (!isMatching) {
+      await this.updateFailedAttempts(user);
+      throw new ApiException('Wrong credentials provided');
+    }
 
     this.analyticsService.upsertUser(user, user._id);
 
+    if (user?.failedLogin?.times > 0) {
+      await this.resetFailedAttempts(user);
+    }
```

Comment:
uummm, we can totally skip the reset counter if we change the logic from `attempt >= 5` to `attempt !== 0 && attempt % 5 === 0` 
wdyt? 🤔

Reply:
@BiswaViraj from the code here I see that there would never be more than 5 failed attempts... 
IMO we should clear it always if there are any failed attempts like the code does

Reply:
> In the current flow, it won't make a DB call unless the user had a failed attempt before the login, If we remove the condition, then it will be called for every single login calls not sure, if it would be beneficial 😕

For security features, right now, I would focus more on solid flows than in performance. We have a lot of room for improvement in the future regarding that. Besides what @LetItRock mentions.